### ISSUE TITLE
Wrong weight matrix

### ISSUE THREAD

**Issue by wxgeo** [wxgeo/smallgraphlib#2]
```python
In [18]: g=WeightedDirectedGraph.from_string("A:B=0.6 B:A=0.9 A:A=0.4 B:B=0.1")

In [19]: print(g.as_tikz())
\providecommand{\contour}[2]{#2}\begin{tikzpicture}[every node/.style = {font={\scriptsize}},vertex/.style = {draw, circle,font={\scriptsize},inner sep=2},directed/.style = {-{Stealth[scale=1.1]}},reversed/.style = {{Stealth[scale=1.1]}-},undirected/.style = {},]
\node[vertex,] (A) at (0.0:1cm) {$A$};
\node[vertex,] (B) at (180.0:1cm) {$B$};
\draw[directed] (A) to [out=-45.0,in=45.0,looseness=5] node[midway] {\contour{white}{0.4}} (A);
\draw[directed] (B) to [out=135.0,in=225.0,looseness=5] node[midway] {\contour{white}{0.1}} (B);
\draw[directed] (A) to[bend left=15] node[pos=0.5] {\contour{white}{0.6}} (B);
\draw[reversed] (A) to[bend right=15] node[pos=0.5] {\contour{white}{0.9}} (B);
\draw[directed] (B) to[bend left=15] node[pos=0.5] {\contour{white}{0.9}} (A);
\draw[reversed] (B) to[bend right=15] node[pos=0.5] {\contour{white}{0.6}} (A);
\end{tikzpicture}

In [20]: g.weight_matrix
Out[20]: ((0, 0.6), (0.9, 0))

In [21]: g
Out[21]: WeightedDirectedGraph(('A', 'B'), ('A', 'A', 0.4), ('A', 'B', 0.6), ('B', 'A', 0.9), ('B', 'B', 0.1))

In [22]: g.weight("A","A")
Out[22]: 0
```

**Comment by wxgeo** [comment]
In fact, `weight` behaviour should differ, based on graph usage:
- for distances:
    - between A and A: weight is always 0; 
    - between A and B: it is `min(weights(A,B))`
    - for missing edges: it is +infinity
- for flow networks:
    - between A and A: weight is always +infinity; 
    - between A and B: it is `sum(weights(A,B))`
    - for missing edges: it is 0
- for Markov chains:
    - between A and A: weight is  `1 - sum(weight(A, Ai) for Ai != A)`. It should be verified at 
    - between A and B: there should be only edge, so its simply edge value
    - for missing edges: it is 0
    
So, the cleaner solution is probably to use several classes, and to not define `weight()` at all (since, after all, weights can be accessed as labels).
Maybe `WeightedGraph` and `WeightedDirectedGraph` should also be removed, and more specific classes used instead.

**Comment by wxgeo** [comment]
Fixed: 72b6a227
